sdk/include/psdk/fltdefs.h

```diff
@@ -0,0 +1,270 @@
+/*
+ * fltdefs.h
+ *
+ * This file is part of the ReactOS PSDK package.
+ *
+ * Contributors:
+ *   Created by Timo Kreuzer.
+ *
+ * THIS SOFTWARE IS NOT COPYRIGHTED
+ *
+ * This source code is offered for use in the public domain. You may
+ * use, modify or distribute it freely.
+ *
+ * This code is distributed in the hope that it will be useful but
+ * WITHOUT ANY WARRANTY. ALL WARRANTIES, EXPRESS OR IMPLIED ARE HEREBY
+ * DISCLAIMED. This includes but is not limited to warranties of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ */
+
+#ifndef _FLTDEFS_H
+#define _FLTDEFS_H
+
+#pragma once
+
+#ifdef __cplusplus
+#define EXTERNCDECL EXTERN_C
+//extern "C" {
+#else
+#define EXTERNCDECL
+#endif
+
+#ifdef _M_CEE_PURE
+#define PFEXPORT
+#else
+#define PFEXPORT __declspec(dllexport)
```

Comment:
I see that this is the same in MS's headers, but this seems wrong. Only the dll itself should try to export these functions, presumably anyone else should use `dllimport` instead.

In any case, this also explains why the functions get exported correctly no matter what the spec file says.

Reply:
I removed the dllexport and now it only exports the correctly annotated functions. Looks like a GCC bug, as on MSVC builds it was correct. Only thing still different from Windows is the order of the exports, but I am unsure how this is supposed to work correctly and why MSVC/Windows have them ordererd based on the function name rather than the export/symbol name.